src/wallet/wallet.cpp

```diff
@@ -4212,7 +4212,7 @@ bool CWallet::HasWalletDescriptor(const WalletDescriptor& desc) const
     return false;
 }
 
-std::shared_ptr<ScriptPubKeyMan> CWallet::AddWalletDescriptor(WalletDescriptor& desc, const FlatSigningProvider& signing_provider, const std::string& label, OutputType& output_type)
+std::shared_ptr<ScriptPubKeyMan> CWallet::AddWalletDescriptor(WalletDescriptor& desc, const FlatSigningProvider& signing_provider, const std::string& label, Optional<OutputType> parsed_output_type)
```

Comment:
`OutputType` needs to be a reference in order for it to work as an out parameter. Should be `Optional<OutputType>&`

Reply:
oops, good catch. I was treating Optional<T> as a pointer but it's actually just a wrapper object. Will update.